### PR TITLE
Add unit tests for iwyu_regex, iwyu_verrs and iwyu_stl_util

### LINKED ISSUE #1994: No unit tests for iwyu_regex module (RegexMatch, RegexReplace, ParseRegexDialect)
The `iwyu_regex` module provides three public functions but has no corresponding unit test file in `unittests/`. The module supports two regex dialects (LLVM and ECMAScript) and is used throughout IWYU for mapping file pattern matching.

Suggested: create `unittests/iwyu_regex_test.cc` with GoogleTest cases covering both dialects, edge cases, and the dialect parser. Register in `CMakeLists.txt`.

### LINKED ISSUE #1996: No unit tests for iwyu_verrs module (SetVerboseLevel, GetVerboseLevel, ShouldPrint)
The `iwyu_verrs` module controls IWYU's logging verbosity and is used extensively via VERRS() and ERRSYM() macros, but has no unit test file.

Suggested: create `unittests/iwyu_verrs_test.cc` with tests for SetVerboseLevel/GetVerboseLevel roundtrip and ShouldPrint logic. Register in `CMakeLists.txt`.

### LINKED ISSUE #2000: iwyu_stl_util.h: ContainsKey, ContainsValue, FindInMap, GetOrDefault lack unit tests
The `iwyu_stl_util.h` header defines several template utility functions but `unittests/iwyu_stl_util_test.cc` only tests Union and UnionSetSemantics. The following functions have no test coverage: ContainsKey, ContainsValue, ContainsKeyValue, ContainsAnyKey, GetOrDefault, FindInMap, RemoveAllFrom.

### PR DESCRIPTION
**Pull request by pierluigilenoci**
This PR adds unit test coverage for three modules that previously had none or incomplete coverage.

### New: `unittests/iwyu_regex_test.cc`
Closes #1994

Tests `ParseRegexDialect`, `RegexMatch`, and `RegexReplace` for both LLVM and ECMAScript dialects, including edge cases (empty patterns, no-match, anchored full-match semantics, capture group replacement).

### New: `unittests/iwyu_verrs_test.cc`
Closes #1996

Tests `SetVerboseLevel`/`GetVerboseLevel` roundtrip and `ShouldPrint` logic at various verbosity levels. Uses a test fixture to save and restore the verbose level between tests.

### Extended: `unittests/iwyu_stl_util_test.cc`
Closes #2000

Adds test cases for `ContainsKey`, `ContainsValue`, `ContainsKeyValue`, `ContainsAnyKey`, `GetOrDefault`, `FindInMap`, and `RemoveAllFrom` — all previously untested.

### CMakeLists.txt
Registers the two new test targets following the existing pattern.

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
Hi @kimgr, @bolshakov-a — this PR adds unit test coverage for `iwyu_regex`, `iwyu_verrs`, and `iwyu_stl_util` (all previously untested or partially tested). Would you be willing to review? Thank you!

File: unittests/iwyu_regex_test.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  RegexDialect dialect = RegexDialect::LLVM;
+  EXPECT_FALSE(ParseRegexDialect("invalid", &dialect));
+  // Dialect should remain unchanged on failure.
+  EXPECT_EQ(RegexDialect::LLVM, dialect);
+}
+
+TEST(IwyuRegexTest, ParseRegexDialectEmpty) {
+  RegexDialect dialect = RegexDialect::LLVM;
+  EXPECT_FALSE(ParseRegexDialect("", &dialect));
+}
```
Comment (kimgr) [inline unittests/iwyu_regex_test.cc]: I feel like these two tests can be merged; empty string is also an invalid value here.

File: unittests/iwyu_regex_test.cc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  EXPECT_FALSE(RegexMatch(RegexDialect::LLVM, "hello", "world"));
+}
+
+TEST(IwyuRegexTest, RegexMatchLLVMPartialDoesNotMatch) {
+  // RegexMatch uses full-match semantics (anchored), so partial patterns
+  // do not match.
+  EXPECT_FALSE(RegexMatch(RegexDialect::LLVM, "hello world", "hello"));
+}
+
+TEST(IwyuRegexTest, RegexMatchLLVMWildcard) {
```
Comment (kimgr) [inline unittests/iwyu_regex_test.cc]: `RegexMatch` and `RegexReplace` are in a sense polymorphic over `RegexDialect`. I see there are repeated tests for `Exact`, `NoMatch`, `PartialDoesNotMatch`, but not `Wildcard`, `EmptyPattern` and `Digits` (not sure, maybe LLVM regexes don't support character classes?).

I think you could use some GTest abstraction feature, probably value-parameterized tests, to run the same testcases over both dialects: http://google.github.io/googletest/advanced.html#value-parameterized-tests. That would also make it clearer what the differences between them are.

File: unittests/iwyu_regex_test.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+TEST(IwyuRegexTest, RegexMatchECMAScriptDigits) {
+  EXPECT_TRUE(RegexMatch(RegexDialect::ECMAScript, "test123", "test[0-9]+"));
+  EXPECT_FALSE(RegexMatch(RegexDialect::ECMAScript, "test", "test[0-9]+"));
+}
+
+TEST(IwyuRegexTest, RegexMatchECMAScriptPartialDoesNotMatch) {
+  EXPECT_FALSE(RegexMatch(RegexDialect::ECMAScript, "hello world", "hello"));
+}
+
+TEST(IwyuRegexTest, RegexReplaceLLVM) {
```
Comment (kimgr) [inline unittests/iwyu_regex_test.cc]: Same for the `Replace` tests, there's a lot of same-behavior between the two dialects and it would be nice to group those together and only keep the differences separate.

File: unittests/iwyu_regex_test.cc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  EXPECT_EQ("hello", result);
+}
+
+TEST(IwyuRegexTest, RegexReplaceECMAScriptWithCapture) {
+  std::string result =
+      RegexReplace(RegexDialect::ECMAScript, "hello world",
+                   "(hello) (world)", "$2 $1");
+  EXPECT_EQ("world hello", result);
+}
+
```
Comment (kimgr) [inline unittests/iwyu_regex_test.cc]: The most important difference between LLVM and ECMAScript is the latter supports negative lookaround (see commit message for cf5388082266e93afe3b3e9688b1bac27319eb89 and a5d8408c5f2cd8579d6cb12fddc7a9727ea83f21, and issues #981 and #935). Would be nice to add tests to highlight that lookahead/lookbehind works with one but not the other.

**Comment by kimgr** [comment]
Thanks! Some suggestions inline.

Could you also squash the commits in the PR, so I can review the commit message as it will appear on master?

Finally, IWYU uses more conversational style for the commit subject, so the [conventional-commits style](https://www.conventionalcommits.org) looks a bit off. Please just update it to `Add unit tests for iwyu_regex, iwyu_verrs and iwyu_stl_util`

File: unittests/iwyu_stl_util_test.cc [hunk 5]
```diff
@@ -53,5 +59,132 @@ TEST(IwyuSTLUtilTest, Union) {
...
+  EXPECT_EQ(set<int>({1, 2, 3}), target);
+}
+
+TEST(IwyuSTLUtilTest, RemoveAllFromEmpty) {
+  set<int> target{1, 2, 3};
+  set<int> to_remove;
+  RemoveAllFrom(to_remove, &target);
+  EXPECT_EQ(set<int>({1, 2, 3}), target);
+}
+
```
Comment (kimgr) [inline unittests/iwyu_stl_util_test.cc:188]: Oh, by the way, what happened to `InsertAllInto`, `Extend` and `GetUniqueEntries`?

File: unittests/iwyu_regex_test.cc (hunk 1 above)
Comment (pierluigilenoci) [reply unittests/iwyu_regex_test.cc]: Done — merged into a single `ParseRegexDialectInvalidOrEmpty` test.

File: unittests/iwyu_regex_test.cc (hunk 2 above)
Comment (pierluigilenoci) [reply unittests/iwyu_regex_test.cc]: Done — introduced `RegexDialectTest` as a parameterized test fixture (`INSTANTIATE_TEST_SUITE_P`) over both dialects. Shared tests cover Exact, NoMatch, PartialDoesNotMatch, Wildcard, EmptyPattern, and Digits. Dialect-specific tests (capture syntax, lookaround) remain separate.

File: unittests/iwyu_regex_test.cc (hunk 3 above)
Comment (pierluigilenoci) [reply unittests/iwyu_regex_test.cc]: Done — ReplaceExact and ReplaceNoMatch are now parameterized alongside the Match tests. Only the capture-group tests remain separate (LLVM uses `\\1`, ECMAScript uses `$1`).

File: unittests/iwyu_regex_test.cc (hunk 4 above)
Comment (pierluigilenoci) [reply unittests/iwyu_regex_test.cc]: Done — added `ECMAScriptNegativeLookahead` and `ECMAScriptNegativeLookbehind` tests to highlight the key LLVM vs ECMAScript difference.

File: unittests/iwyu_stl_util_test.cc (hunk 5 above)
Comment (pierluigilenoci) [reply unittests/iwyu_stl_util_test.cc:188]: Added! Tests for `InsertAllInto`, `Extend`, and `GetUniqueEntries` are now in iwyu_stl_util_test.cc. Also squashed into a single commit with the conversational title as requested.

File: unittests/iwyu_regex_test.cc (hunk 4 above)
Comment (pierluigilenoci) [reply unittests/iwyu_regex_test.cc]: Investigated further: `std::regex` ECMAScript mode follows the ES5 spec which only supports lookahead (`(?=...)` and `(?!...)`), not lookbehind (`(?<=...)` and `(?<!...)`) — lookbehind was added in ES2018. The `ECMAScriptNegativeLookbehind` test was causing the CI abort (SIGABRT from `std::regex_error`). Removed it and updated the comment to clarify this limitation. The `ECMAScriptNegativeLookahead` test is kept as it correctly demonstrates the key LLVM-vs-ECMAScript difference.

File: unittests/iwyu_regex_test.cc (hunk 2 above)
Comment (kimgr) [reply unittests/iwyu_regex_test.cc]: Thanks!

File: unittests/iwyu_regex_test.cc (hunk 1 above)
Comment (kimgr) [reply unittests/iwyu_regex_test.cc]: Nice, thanks

File: unittests/iwyu_regex_test.cc (hunk 4 above)
Comment (kimgr) [reply unittests/iwyu_regex_test.cc]: Oh, interesting! I had just assumed lookbehind would also be supported by `std::regex`. Good to have more empirical info.
